cli/link.go

```diff
@@ -84,17 +86,47 @@ func (p PipelineLink) IsLocal() bool {
 	return p.FsAllow
 }
 
-//checks if the pipeline is up
-//otherwise it brings it up and fills the
-//link object
+// checks if the pipeline is up
+// otherwise it brings it up and fills the
+// link object
 func bringUp(pLink *PipelineLink) error {
 	alive, err := pLink.pipeline.Alive()
 	if err != nil {
-		if pLink.config[STARTING].(bool) {
-			alive, err = NewPipelineLauncher(pLink.pipeline,
-				pLink.config.ExecPath(), pLink.config[TIMEOUT].(int)).Launch(os.Stdout)
-			if err != nil {
-				return fmt.Errorf("Error bringing the pipeline2 up %v", err.Error())
+		// Requested behavior :
+		// dp2 launches pipeline-ui (if needed); dp2 calls pipeline-ui as cli handler; pipeline-ui calls dp2 with the port info as an argument
+
+		// When port attribute is set, the cli tool should not try to start a pipeline instance
+		// and directly connect to pipeline instance for which the port is set on the command line
+		if pLink.config[STARTING].(bool) && !(slices.Contains(os.Args[1:], "--port") || slices.Contains(os.Args[1:], "--host")) {
+			execpath := pLink.config.ExecPath()
+			// if execpath ends with DAISY Pipeline or DAISY Pipeline.exe
+			// we forward the command to the electron app
+			if strings.HasSuffix(execpath, "DAISY Pipeline") || strings.HasSuffix(execpath, "DAISY Pipeline.exe") {
+				args := os.Args[1:]
+				if len(os.Args) == 1 {
+					args = append(args, "help")
+				}
+				absPath, err := exec.LookPath(execpath)
+				if err != nil {
+					absPath = execpath
+				}
+				cmd := exec.Command(absPath, args...)
+				cmd.Stdout = os.Stdout
+				cmd.Stderr = os.Stderr
+				if err := cmd.Run(); err != nil {
+					return fmt.Errorf("Could not use the pipeline app: %v", err)
+				} else {
+					os.Exit(cmd.ProcessState.ExitCode())
+				}
```

Comment:
Is there nothing like the [`exec()`](https://man7.org/linux/man-pages/man3/exec.3.html) function in Go?

Reply:
Hum not sure

(Note that I don't think exec has an equivalent on windows to "replace" an ongoing executable process, so maybe they cannot make a replicate of that in go)

Reply:
Ah yes, that could be the reason.